src/app/authorisation/auth.guard.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  CanActivate,
+  Router,
+  RouterStateSnapshot,
+  UrlTree,
+} from '@angular/router';
+import { Observable } from 'rxjs';
+import { AuthService } from './auth.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthGuard implements CanActivate {
+  constructor(private authService: AuthService, private router: Router) {}
+  canActivate() {
```

Comment:
Just a note - class based guards are deprecated since Angular 15, and now it's recommended to generate functional guards (a function using `inject()`, instead of a class https://angular.io/guide/router#preventing-unauthorized-access. No need to change this now, but good to know for the future.